aws-crt/lib/aws-crt/io.rb

```diff
@@ -0,0 +1,23 @@
+module Aws::Crt
+  module IO
+    # A collection of event-loops.
+    # An event-loop is a thread for doing async work, such as I/O.
+    # Classes that need to do async work will ask the EventLoopGroup for an event-loop to use.
+    class EventLoopGroup
+      def initialize(max_threads = nil)
+        unless max_threads.nil? || (max_threads.is_a?(Integer) && max_threads.positive?)
+          raise ArgumentError, 'max_threads must be nil or positive Integer'
+        end
+
+        max_threads = 0 if max_threads.nil?
```

Comment:
to be more explicit here I'd create a constant DEFAULT for zero

Reply:
added comment to make this more clear